server/src/middleware/mod.rs

```diff
@@ -1,4 +1,9 @@
 //! Various middleware implementations for RPC specific purposes.
 
+/// HTTP Host filtering middleware.
+mod host_filter;
 /// Proxy `GET /path` to internal RPC methods.
-pub mod proxy_get_request;
+mod proxy_get_request;
+
+pub use host_filter::*;
+pub use proxy_get_request::*;
```

Comment:
Nit: Given that `host_filter` has types like `Authority` etc in, maybe it's best to expose each one in its own module so it's obvious which types are for what (especially if we ever add any more)?

Reply:
Added an `uri mod` and simplified the `host_filter mod`, lemme know if you are happy with it? :)

Reply:
Def looks better now that the name `Host` is in all of the exposed things! Personally I liked that the `Authority` stuff lived here (because it's only used in this filter so keeping the code next to it felt good) but I'm easy either way :)